src/engraving/libmscore/beam.cpp

```diff
@@ -318,11 +318,22 @@ void Beam::layout1()
         } else if (_isGrace) {
             _up = true;
         } else {
+            // search through the beam for the first chord with explicit stem direction and use that.
+            // if there is no explicit stem direction, default to the direction of the first stem.
+            bool firstUp = false;
+            bool firstChord = true;
+            DirectionV explicitDirection = DirectionV::AUTO;
             for (ChordRest* cr :_elements) {
                 if (cr->isChord()) {
-                    _up = cr->up();
-                    break;
+                    if (toChord(cr)->stemDirection() != DirectionV::AUTO) {
```

Comment:
Please, never call the same operations multiple times even if they are relatively fast to perform. Call this stemDirection() once, write its result to a temp variable, and reuse it